node/modules/client.go

```diff
@@ -173,13 +151,13 @@ func NewClientDatastore(ds dtypes.MetadataDS) dtypes.ClientDatastore {
 	return namespace.Wrap(ds, datastore.NewKey("/deals/client"))
 }
 
-func StorageClient(lc fx.Lifecycle, h host.Host, ibs dtypes.ClientBlockstore, mds dtypes.ClientMultiDstore, r repo.LockedRepo, dataTransfer dtypes.ClientDataTransfer, discovery *discoveryimpl.Local, deals dtypes.ClientDatastore, scn storagemarket.StorageClientNode, j journal.Journal) (storagemarket.StorageClient, error) {
+func StorageClient(lc fx.Lifecycle, h host.Host, dataTransfer dtypes.ClientDataTransfer, discovery *discoveryimpl.Local, deals dtypes.ClientDatastore, scn storagemarket.StorageClientNode, j journal.Journal) (storagemarket.StorageClient, error) {
```

Comment:
Hum, do they? Are they instantiating Lotus programmatically and overriding this with their own blockstore through DI? Do we have concrete users that use this? It's a very hacky thing to tap into implementation details through DI to serve an external use case (we should be having a dedicated API for this if we expect this case to be supported externally going forward).

We thought this blockstore was used only internally. We absolutely don't want to break existing clients. Can we identify who these clients are?

Reply:
@raulk I believe Hannah is referring to this: https://github.com/filecoin-project/lotus/blob/v1.11.1-rc2/node/builder_chain.go#L169-L174

Textile virtually never uses the lotus-local blockstore, and @dkkapur and I use the IPFS linkage in various cases for slingshot work ( the flow is `lotus client retrieve ... /dev/null` and then pulling whatever one needs from the IPFS instance )

Reply:
Original context: https://github.com/filecoin-project/lotus/pull/1843

Reply:
Thoughts here after discussing with @dirkmc off band:

1. We attempt to make the go-fil-markets API deal with blockstores only for storage and retrievals, removing all path and CAR file management from that library. This was already identified in previous reviews as desirable.
2. This means removing all CAR paths from deal state objects in go-fil-markets.
3. This complicates resumption on restart; before go-fil-markets used to have the CAR path so it could resume autonomously. Now it will need to ask to reopen the blockstore through some interface (possibly a retrieval/storage node adaptor, or the existing getter/fetcher?).
4. When the node is configured to use the IPFS Blockstore, the interface would always return the blockstore instead of CAR files.
    - We assume that `lotus client retrieve` calls with an `<output_path>` do not use the path in this case? Is this even handled now? Calls with an output path should be rejected.
5. To avoid breakage of existing clients, `--car` should unpack the CARv1 from within the CARv2. Right now we'll need to do this through an inefficient copy, until https://github.com/ipld/go-car/issues/207 is implemented.
    - Please add a TODO in code pointing to that issue.
    - Alternatively, we could just choose to break backwards compatibility and return a CARv2.
    - If we choose not to, we should add a `--carv2` flag that just returns a CARv2.

@dirkmc ^^ can you confirm if this captures what we discussed?

Reply:
@dirkmc ^^ since this got implemented quickly, are we ready to adopt it?

Reply:
Fixed.